autoload/OmniSharp/stdio.vim

```diff
@@ -86,31 +80,55 @@ function! s:Request(command, opts) abort
   let buffer = join(lines, sep)
 
   let body = {
-  \ 'Seq': s:nextseq,
-  \ 'Command': a:command,
-  \ 'Type': 'Request',
   \ 'Arguments': {
   \   'Filename': filename,
   \   'Line': lnum,
   \   'Column': cnum,
   \   'Buffer': buffer
   \ }
   \}
-  if has_key(a:opts, 'Parameters')
-    call extend(body.Arguments, a:opts.Parameters, 'force')
+  call s:RawRequest(body, a:command, a:opts, sep)
+endfunction
+
+" body -> request body dict
+" command -> command string. eg: /codecheck
+" opts -> command arugments dict
+" sep -> line separator. Optional, if not given line ending replacemment is not done
+function! s:RawRequest(...) abort
+  let body = get(a:, 1, 0)
+  let command = get(a:, 2, 0)
+  let opts = get(a:, 3, 0)
+  let sep = get(a:, 4, 0)
+
+  let job = OmniSharp#GetHost().job
+  if type(job) != type({}) || !has_key(job, 'job_id') || !job.loaded
+    return 0
+  endif
+  let job_id = job.job_id
+  call s:Log(job_id . '  Request: ' . command, 'debug')
+
+  let body['Command'] = command
+  let body['Seq'] = s:nextseq
+  let body['Type'] = 'Request'
+  if has_key(opts, 'Parameters')
+    call extend(body.Arguments, opts.Parameters, 'force')
+  endif
+  if type(sep) == type('')
```

Comment:
If we set the default value for `sep` to `''` as in my comment above, we can just test against that, instead of comparing types:

```vim
  if sep !=# ''
```

Reply:
Done